src/frontend/src/lib/services/address.services.ts

```diff
@@ -157,23 +166,47 @@ export const certifyAddress = async (
 			return { success: false, err: 'Using the dapp with an anonymous user if not supported.' };
 		}
 
-		const certifiedAddress = await getEthAddress(identity);
+		const certifiedAddress = await getAddress(identity);
 
 		if (address.toLowerCase() !== certifiedAddress.toLowerCase()) {
 			return {
 				success: false,
-				err: 'The address used to load the data did not match your actual wallet address, which is why your session was ended. Please sign in again to reload your own data.'
+				err: `The address used to load the data did not match your actual ${tokenId.description} wallet address, which is why your session was ended. Please sign in again to reload your own data.`
```

Comment:
I guess it's already the plan but, just in case, could you extract this i18n in another PR?

Reply:
indeed I was going to do it in another PR, since there are actually more eheh